src/rgw/driver/rados/rgw_rados.cc

```diff
@@ -4199,6 +4199,9 @@ int RGWRados::fetch_remote_obj(RGWObjectCtx& dest_obj_ctx,
   obj_time_weight set_mtime_weight;
   set_mtime_weight.high_precision = high_precision_time;
   int ret;
+  auto get_obj_name = [&src_obj, &dest_obj]() -> std::string {
+    return (std::stringstream{} << "object(src=" << src_obj << ", dest=" << dest_obj << ")").str();
+  };
```

Comment:
This construction is very strange. If src_obj and dest_obj aren't going to change, this should just be a const string. If they are, they would be better as parameters so it's obvious.

Also use `fmt::format` rather than stringstream, something like:
```
      return fmt::format("object(src={}, dest={})", src_obj, dest_obj);
```

Reply:
>  If src_obj and dest_obj aren't going to change, this should just be a const string. 

Yes, since the captured variables `src_obj` and `dest_obj` are const `rgw_obj`s, added the `const` qualifier to the return type.

> Also use fmt::format rather than stringstream, something like ...

There's `operator<<` overload for `rgw_obj` but no `Formatter` specialization so the suggested way fails to compile but used `fmt::format` in the latest evidence and explicitly logged `bucket_id` and `object key` (similar to what `<<` overload does).

Re-tested and updated the test results in the Description section.

Reply:
Hey @adamemerson - did you get a chance to look at the latest evidence? Just made the obj_name being replicated a constant.